py/server/deephaven/table.py

```diff
@@ -1451,3 +1451,49 @@ def get_constituent(self, key_values: Union[Any, Sequence[Any]]) -> Optional[Tab
     def constituent_tables(self) -> List[Table]:
         """Returns all the current constituent tables."""
         return list(map(Table, self.j_partitioned_table.constituents()))
+
+    def transform(self, func: Callable[[Table], Table]) -> PartitionedTable:
+        """Apply the provided function to all constituent Tables and produce a new PartitionedTable with the results
+        as its constituents, with the same data for all other columns in the underlying partitioned Table. Note that
+        if the Table underlying this PartitionedTable changes, a corresponding change will propagate to the result.
+
+        Args:
+            func (Callable[[Table], Table]: a function which takes a Table as input and returns a new Table
+
+        Returns:
+            a PartitionedTable
+
+        Raises:
+            DHError
+        """
+        try:
+            j_operator = j_unary_operator(func, dtypes.from_jtype(Table.j_object_type.jclass))
+            j_pt = self.j_partitioned_table.transform(j_operator)
+            return PartitionedTable(j_partitioned_table=j_pt)
+        except Exception as e:
+            raise DHError(e, "failed to transform the PartitionedTable.") from e
+
+    def partitioned_transform(self, other: PartitionedTable, func: Callable[[Table, Table], Table]) -> PartitionedTable:
+        """Join the underlying partitioned Tables from this PartitionedTable and other on the key columns, then apply
+        the provided function to all pairs of constituent Tables with the same keys in order to produce a new
+        PartitionedTable with the results as its constituents, with the same data for all other columns in the
+        underlying partitioned Table from this. Note that if the Tables underlying this PartitionedTable or other
+        change, a corresponding change will propagate to the result.
```

Comment:
"""Join this and the other PartitionedTables on the key columns and applies the provided function to all pairs
        of constituent tables with the same keys.  The resulting tables are used to produce a new PartitionedTable 
        with the results as its constituents and with the same data as this PartitionedTable for all other columns.

        Note that if the Tables underlying this PartitionedTable or other change, a corresponding change will 
        propagate to the result.